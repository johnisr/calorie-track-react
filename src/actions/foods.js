import uuid from 'uuid';
import database from '../firebase/firebase';
import moment from 'moment';

export const addFood = food => ({
  type: 'ADD_FOOD',
  food: { id: uuid(), ...food },
});

export const startAddFood = (data = {}) => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;
    const {
      name = '',
      amount = '0',
      unit = 'grams',
      carbohydrates = 0,
      protein = 0,
      fat = 0,
      calories = 0,
      timesUsed = 0,
      createdAt = moment(),
    } = data;
    const food = { name, amount, unit, carbohydrates, protein, fat, calories, timesUsed, createdAt };
    try {
      const res = await database.ref(`users/${uid}/foods`).push(food);
      dispatch(addFood({ id: res.key, ...food }));
    } catch (e) {

    }
  };
}

export const editFood = (id, updates) => ({
  type: 'EDIT_FOOD',
  id,
  updates,
});

export const startEditFood = (id, updates) => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;
    try {
      await database.ref(`users/${uid}/foods/${id}`).update(updates);
      dispatch(editFood(id, updates));
    } catch (e) {
      
    }
  };
};

export const removeFood = ({ id }) => ({
  type: 'REMOVE_FOOD',
  id,
});

export const startRemoveFood = ({ id }) => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;
    try {
      await database.ref(`users/${uid}/foods/${id}`).remove();
      dispatch(removeFood({ id }));
    } catch (e) {
      
    }
  };
};

export const setFoods = foods => ({
  type: 'SET_FOODS',
  foods,
});

export const startSetFoods = () => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;

    try {
      const snapshot = await database.ref(`users/${uid}/foods`).once('value');
      const foods = [];
      snapshot.forEach((food) => {
        foods.push({
          id: food.key,
          ...food.val(),
        });
      });
      dispatch(setFoods(foods));
    } catch (e) {

    }
  };
};