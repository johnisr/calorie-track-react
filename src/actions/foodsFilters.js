export const setFoodNameFilter = (name = '') => ({
  type: 'SET_FOOD_NAME_FILTER',
  name,
});

export const sortByUsage = () => ({
  type: 'SORT_BY_USAGE',
});

export const sortByDate = () => ({
  type: 'SORT_BY_DATE',
});

export const setMaxFoodsShown = (max = 10) => ({
  type: 'SET_MAX_FOODS_SHOWN',
  max,
});

export const setOffset = (offset = 0) => ({
  type: 'SET_OFFSET',
  offset,
});
