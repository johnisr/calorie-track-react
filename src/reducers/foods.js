const foodsDefaultState = [];

export default (state = foodsDefaultState, action) => {
  switch (action.type) {
    case 'ADD_FOOD':
      return [...state, action.food];
    case 'EDIT_FOOD':
      return state.map((food) => {
        if (food.id === action.id) {
          return {
            ...food,
            ...action.updates,
          };
        }
        return food;
      });
    case 'REMOVE_FOOD':
      return state.filter(({ id }) => id !== action.id);
    case 'SET_FOODS':
      return action.foods;
    default:
      return state;
  }
}