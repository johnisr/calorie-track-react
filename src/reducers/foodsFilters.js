
const foodsFilterReducerDefaultState = {
  name: '',
  sortBy: 'usage',
  maxFoodsShown: 10,
  offset: 0,
};

export default (state = foodsFilterReducerDefaultState, action) => {
  switch(action.type) {
    case 'SET_FOOD_NAME_FILTER': {
      return {
        ...state,
        name: action.name,
      };
    }
    case 'SORT_BY_USAGE': {
      return {
        ...state,
        sortBy: 'usage',
      };
    }
    case 'SORT_BY_DATE': {
      return {
        ...state,
        sortBy: 'date',
      };
    }
    case 'SET_MAX_FOODS_SHOWN': {
      return {
        ...state,
        maxFoodsShown: action.max,
      };
    }
    case 'SET_FOODS_OFFSET': {
      return {
        ...state,
        offset: action.offset,
      };
    }

    default:
      return state;
  }
};