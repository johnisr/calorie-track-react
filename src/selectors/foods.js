export default (foods, { name, sortBy, maxFoodsShown, offset }) => {
  return foods
    .filter((food) => {
      const nameMatch = food.name.toLowerCase().includes(name.toLowerCase());
      return nameMatch;
    })
    .sort((a, b) => {
      if (sortBy === 'usage') {
        return a.timesUsed < b.timesUsed ? 1 : -1;
      } else if (sortBy === 'date') {
        return a.createdAt < b.createdAt ? 1 : -1;
      }
    })
    .slice((offset * maxFoodsShown), (offset * maxFoodsShown) + maxFoodsShown);
}