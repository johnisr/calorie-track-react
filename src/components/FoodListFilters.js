import React from 'react';
import { connect } from 'react-redux';
import { setFoodNameFilter, sortByUsage, sortByDate, setMaxFoodsShown, setFoodsOffset } from '../actions/foodsFilters';
import selectFoods from '../selectors/foods';

export class FoodListFilters extends React.Component {
  handleNameChange = (e) => {
    this.props.setFoodNameFilter(e.target.value);
    this.props.setFoodsOffset(0);
  };
  handleSortChange = (e) => {
    if (e.target.value === 'usage') {
      this.props.sortByUsage();
    } else if (e.target.value === 'date') {
      this.props.sortByDate();
    }
  };
  handleMaxFoodPageChange = (e) => {
    this.props.setMaxFoodsShown(parseInt(e.target.value, 10));
    this.props.setFoodsOffset(0);
  };
  handlePrevPageChange = () => {
    this.props.setFoodsOffset(this.props.filters.offset - 1);
  };
  handleNextPageChange = () => {
    this.props.setFoodsOffset(this.props.filters.offset + 1);
  };
  handleOffsetChange = (i) => {
    this.props.setFoodsOffset(i);
  }

  createPageButtons = () => {
    const foodsLength = this.props.foodsLength;
    const offset = this.props.filters.offset;
    const maxFoodsShown = this.props.filters.maxFoodsShown;

    const pagesNeeded = foodsLength % maxFoodsShown === 0 ?
      foodsLength / maxFoodsShown :
      (foodsLength / maxFoodsShown) + 1;
    
    let start;
    if (pagesNeeded - offset < 5) { // reaching end of results 
      start = Math.max(0, pagesNeeded - 10);
    } else { // Normal case
      start = Math.max(0, offset - 5);
    }

    let end;
    if (start === 0) { // pages 1 .. (pagesNeeded or 10)
      end = Math.min(pagesNeeded - 1, 10); // count starts from 0
    } else { // Normal case
      end = Math.min(offset + 5, pagesNeeded);
    }

    const pageButtons = [];
    for (let i = start; i < end; i++) {
      pageButtons.push(
        <button
        key={i}
        onClick={() => this.handleOffsetChange(i)}
        disabled={i === this.props.filters.offset}
        >{i + 1}</button>)
    }
    return pageButtons;
  }
  
  render() {
    const foodsLength = this.props.foodsLength;
    const offset = this.props.filters.offset;
    const maxFoodsShown = this.props.filters.maxFoodsShown;

    const isLastPage = (offset * maxFoodsShown) + maxFoodsShown >= foodsLength;

    return (
      <div>
        <input 
          type="text"
          placeholder="Search Foods"
          value={this.props.filters.name} 
          onChange={this.handleNameChange}
        />
        <select
          className="select"
          value={this.props.filters.sortBy} 
          onChange={this.handleSortChange}
        >
          <option value="usage">Usage</option>
          <option value="date">Date</option>
        </select>
        <select
          className="select"
          value={this.props.filters.maxFoodsShown} 
          onChange={this.handleMaxFoodPageChange}
        >
          <option value={10}>10</option>
          <option value={25}>25</option>
          <option value={50}>50</option>
        </select>
        <button 
          onClick={this.handlePrevPageChange}
          disabled={this.props.filters.offset <= 0}
        >
          Prev Page
        </button>
        {this.createPageButtons()}
        <button 
          onClick={this.handleNextPageChange}
          disabled={isLastPage}
        >
          Next Page
        </button>
      </div>
    );
  };
}

const mapStateToProps = state => {
  const visibleFoods = selectFoods(state.foods, state.foodsFilters);
  return {
    filters: state.foodsFilters,
    foodsLength: visibleFoods.length,
  }
};

const mapDispatchToProps = (dispatch) => ({
  setFoodNameFilter: (name) => dispatch(setFoodNameFilter(name)),
  sortByUsage: () => dispatch(sortByUsage()),
  sortByDate: () => dispatch(sortByDate()),
  setMaxFoodsShown: (max) => dispatch(setMaxFoodsShown(max)),
  setFoodsOffset: (offset) => dispatch(setFoodsOffset(offset)),
});

export default connect(mapStateToProps, mapDispatchToProps)(FoodListFilters);