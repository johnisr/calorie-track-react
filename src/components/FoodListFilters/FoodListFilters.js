import React from 'react';
import { connect } from 'react-redux';
import { setFoodNameFilter, sortByUsage, sortByDate, setMaxFoodsShown, setFoodsOffset } from '../../actions/foodsFilters';
import selectFoods from '../../selectors/foods';

export class FoodListFilters extends React.Component {
  handleNameChange = (e) => {
    this.props.setFoodNameFilter(e.target.value);
    this.props.setFoodsOffset(0);
  };
  handleSortChange = (e) => {
    if (e.target.value === 'usage') {
      this.props.sortByUsage();
    } else if (e.target.value === 'date') {
      this.props.sortByDate();
    }
  };
  handleMaxFoodPageChange = (e) => {
    this.props.setMaxFoodsShown(parseInt(e.target.value, 10));
    this.props.setFoodsOffset(0);
  };
  handlePrevPageChange = () => {
    this.props.setFoodsOffset(this.props.filters.offset - 1);
  };
  handleNextPageChange = () => {
    this.props.setFoodsOffset(this.props.filters.offset + 1);
  };
  handleOffsetChange = (i) => {
    this.props.setFoodsOffset(i);

  }

  createPageButtons = () => {
    const foodsLength = this.props.foodsLength;
    const offset = this.props.filters.offset;
    const maxFoodsShown = this.props.filters.maxFoodsShown;

    const pagesNeeded = foodsLength % maxFoodsShown === 0 ?
      foodsLength / maxFoodsShown :
      (foodsLength / maxFoodsShown) + 1;
    
    let start;
    if (pagesNeeded - offset < 3) { // reaching end of results 
      start = Math.max(0, pagesNeeded - 6);
    } else { // Normal case
      start = Math.max(0, offset - 3);
    }

    let end;
    if (start === 0) { // pages 1 .. (pagesNeeded or 10)
      end = Math.min(pagesNeeded - 1, 6); // count starts from 0
    } else { // Normal case
      end = Math.min(offset + 3, pagesNeeded);
    }

    const pageButtons = [];
    for (let i = start; i < end; i++) {
      pageButtons.push(
        <button
        className={this.props.filters.offset === i ? 'btn btn--page btn--active' : 'btn btn--page'}
        key={i}
        onClick={() => this.handleOffsetChange(i)}
        disabled={i === this.props.filters.offset}
        >{i + 1}</button>)
    }
    return pageButtons;
  }
  
  render() {
    const foodsLength = this.props.foodsLength;
    const offset = this.props.filters.offset;
    const maxFoodsShown = this.props.filters.maxFoodsShown;

    const isLastPage = (offset * maxFoodsShown) + maxFoodsShown >= foodsLength;

    return (
      <div className="foodsFilter">
        <div className="foodsFilter__search">
          <input 
            className="foodsFilter__input"
            type="text"
            placeholder="Search Foods"
            value={this.props.filters.name} 
            onChange={this.handleNameChange}
          />
        </div>
        <div className="foodsFilter__options">
          <select
            className="foodsFilter__select"
            value={this.props.filters.sortBy} 
            onChange={this.handleSortChange}
          >
            <option value="usage">Usage</option>
            <option value="date">Date</option>
          </select>
          <select
            className="foodsFilter__select"
            value={this.props.filters.maxFoodsShown} 
            onChange={this.handleMaxFoodPageChange}
          >
            <option value={1}>1</option>
            <option value={10}>10</option>
            <option value={25}>25</option>
            <option value={50}>50</option>
          </select>        
        </div>
        <div className="foodsFilter__pages flex-right-end">
          <button
            className={offset <= 0 ? "btn btn--page btn--disabled" : "btn btn--page"}
            onClick={this.handlePrevPageChange}
            disabled={offset <= 0}
          >
            Prev
          </button>
          {this.createPageButtons()}
          <button
            className={isLastPage ? "btn btn--page btn--disabled" : "btn btn--page"}
            onClick={this.handleNextPageChange}
            disabled={isLastPage}
          >
            Next
          </button>
        </div>
      </div>
    );
  };
}

const mapStateToProps = state => {
  const visibleFoods = selectFoods(state.foods, state.foodsFilters);
  return {
    filters: state.foodsFilters,
    foodsLength: visibleFoods.length,
  }
};

const mapDispatchToProps = (dispatch) => ({
  setFoodNameFilter: (name) => dispatch(setFoodNameFilter(name)),
  sortByUsage: () => dispatch(sortByUsage()),
  sortByDate: () => dispatch(sortByDate()),
  setMaxFoodsShown: (max) => dispatch(setMaxFoodsShown(max)),
  setFoodsOffset: (offset) => dispatch(setFoodsOffset(offset)),
});

export default connect(mapStateToProps, mapDispatchToProps)(FoodListFilters);