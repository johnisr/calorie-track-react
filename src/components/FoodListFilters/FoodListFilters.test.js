import React from 'react';
import { shallow } from 'enzyme';
import { FoodListFilters } from './FoodListFilters';
import { foodsFilters } from '../../tests/fixtures/foodsFilters';
import foods from '../../tests/fixtures/foods';

let filters, foodsLength, setFoodNameFilter, sortByUsage, sortByDate, setMaxFoodsShown, setFoodsOffset, wrapper;
beforeEach(() => {
  filters = foodsFilters;
  foodsLength = foods.length;
  setFoodNameFilter = jest.fn();
  sortByUsage = jest.fn();
  sortByDate = jest.fn();
  setMaxFoodsShown = jest.fn();
  setFoodsOffset = jest.fn();
  wrapper = shallow(<FoodListFilters 
    filters={filters}
    foodsLength={foodsLength}
    setFoodNameFilter={setFoodNameFilter}
    sortByUsage={sortByUsage}
    sortByDate={sortByDate}
    setMaxFoodsShown={setMaxFoodsShown}
    setFoodsOffset={setFoodsOffset}
  />);
});

it('should render FoodListFilters correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should handle name change', () => {
  const value = 'some value';
  const e = {target: { value }};
  wrapper.find('input').at(0).simulate('change', e);
  expect(setFoodNameFilter).toHaveBeenLastCalledWith(value);
  expect(setFoodsOffset).toHaveBeenLastCalledWith(0);
});

it('should handle sort change to usage', () => {
  const value = 'usage';
  const e = {target: { value }};
  wrapper.find('select').at(0).simulate('change', e);
  expect(sortByUsage).toHaveBeenCalled();
});

it('should handle sort change to date', () => {
  const value = 'date';
  const e = {target: { value }};
  wrapper.find('select').at(0).simulate('change', e);
  expect(sortByDate).toHaveBeenCalled();
});

it('should handle max foods shown', () => {
  const value = '25';
  const e = {target: { value }};
  wrapper.find('select').at(1).simulate('change', e);
  expect(setMaxFoodsShown).toHaveBeenLastCalledWith(25);
});

it('should handle prev page click', () => {
  wrapper.find('button').at(0).simulate('click');
  expect(setFoodsOffset).toHaveBeenLastCalledWith(filters.offset - 1);
});

it('should handle next page click', () => {
  wrapper.find('button').at(2).simulate('click');
  expect(setFoodsOffset).toHaveBeenLastCalledWith(filters.offset + 1);
});

// Buttons [Prev Page] [ 1 ] [Next Page]
it('should handle page Number click', () => {
  wrapper.find('button').at(1).simulate('click');
  expect(setFoodsOffset).toHaveBeenLastCalledWith(0);
});

