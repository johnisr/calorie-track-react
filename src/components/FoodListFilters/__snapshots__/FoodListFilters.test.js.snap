// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FoodListFilters correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FoodListFilters
    filters={
        Object {
            "maxFoodsShown": 10,
            "name": "",
            "offset": 0,
            "sortBy": "usage",
          }
    }
    foodsLength={3}
    setFoodNameFilter={[Function]}
    setFoodsOffset={[Function]}
    setMaxFoodsShown={[Function]}
    sortByDate={[Function]}
    sortByUsage={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="foodsListFilters__search"
>
          <input
                    className="foodsListFilters__input"
                    onChange={[Function]}
                    placeholder="Search Foods"
                    type="text"
                    value=""
          />
</div>,
        <div
          className="foodsListFilters__options"
>
          <select
                    className="foodsListFilters__select"
                    onChange={[Function]}
                    value="usage"
          >
                    <option
                              value="usage"
                    >
                              Usage
                    </option>
                    <option
                              value="date"
                    >
                              Date
                    </option>
          </select>
          <select
                    className="foodsListFilters__select"
                    onChange={[Function]}
                    value={10}
          >
                    <option
                              value={1}
                    >
                              1
                    </option>
                    <option
                              value={10}
                    >
                              10
                    </option>
                    <option
                              value={25}
                    >
                              25
                    </option>
                    <option
                              value={50}
                    >
                              50
                    </option>
          </select>
</div>,
        <div
          className="foodsListFilters__pages flex-right-end"
>
          <button
                    className="btn btn--page btn--disabled"
                    disabled={true}
                    onClick={[Function]}
          >
                    Prev
          </button>
          <button
                    className="btn btn--page btn--active"
                    disabled={true}
                    onClick={[Function]}
          >
                    1
          </button>
          <button
                    className="btn btn--page btn--disabled"
                    disabled={true}
                    onClick={[Function]}
          >
                    Next
          </button>
</div>,
      ],
      "className": "foodsListFilters",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="foodsListFilters__input"
            onChange={[Function]}
            placeholder="Search Foods"
            type="text"
            value=""
/>,
          "className": "foodsListFilters__search",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "foodsListFilters__input",
            "onChange": [Function],
            "placeholder": "Search Foods",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <select
              className="foodsListFilters__select"
              onChange={[Function]}
              value="usage"
>
              <option
                            value="usage"
              >
                            Usage
              </option>
              <option
                            value="date"
              >
                            Date
              </option>
</select>,
            <select
              className="foodsListFilters__select"
              onChange={[Function]}
              value={10}
>
              <option
                            value={1}
              >
                            1
              </option>
              <option
                            value={10}
              >
                            10
              </option>
              <option
                            value={25}
              >
                            25
              </option>
              <option
                            value={50}
              >
                            50
              </option>
</select>,
          ],
          "className": "foodsListFilters__options",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value="usage"
>
                  Usage
</option>,
                <option
                  value="date"
>
                  Date
</option>,
              ],
              "className": "foodsListFilters__select",
              "onChange": [Function],
              "value": "usage",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Usage",
                  "value": "usage",
                },
                "ref": null,
                "rendered": "Usage",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Date",
                  "value": "date",
                },
                "ref": null,
                "rendered": "Date",
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value={1}
>
                  1
</option>,
                <option
                  value={10}
>
                  10
</option>,
                <option
                  value={25}
>
                  25
</option>,
                <option
                  value={50}
>
                  50
</option>,
              ],
              "className": "foodsListFilters__select",
              "onChange": [Function],
              "value": 10,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1",
                  "value": 1,
                },
                "ref": null,
                "rendered": "1",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "10",
                  "value": 10,
                },
                "ref": null,
                "rendered": "10",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "25",
                  "value": 25,
                },
                "ref": null,
                "rendered": "25",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "50",
                  "value": 50,
                },
                "ref": null,
                "rendered": "50",
                "type": "option",
              },
            ],
            "type": "select",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="btn btn--page btn--disabled"
              disabled={true}
              onClick={[Function]}
>
              Prev
</button>,
            Array [
              <button
                className="btn btn--page btn--active"
                disabled={true}
                onClick={[Function]}
>
                1
</button>,
            ],
            <button
              className="btn btn--page btn--disabled"
              disabled={true}
              onClick={[Function]}
>
              Next
</button>,
          ],
          "className": "foodsListFilters__pages flex-right-end",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Prev",
              "className": "btn btn--page btn--disabled",
              "disabled": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Prev",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": 1,
              "className": "btn btn--page btn--active",
              "disabled": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 1,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Next",
              "className": "btn btn--page btn--disabled",
              "disabled": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Next",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="foodsListFilters__search"
>
            <input
                        className="foodsListFilters__input"
                        onChange={[Function]}
                        placeholder="Search Foods"
                        type="text"
                        value=""
            />
</div>,
          <div
            className="foodsListFilters__options"
>
            <select
                        className="foodsListFilters__select"
                        onChange={[Function]}
                        value="usage"
            >
                        <option
                                    value="usage"
                        >
                                    Usage
                        </option>
                        <option
                                    value="date"
                        >
                                    Date
                        </option>
            </select>
            <select
                        className="foodsListFilters__select"
                        onChange={[Function]}
                        value={10}
            >
                        <option
                                    value={1}
                        >
                                    1
                        </option>
                        <option
                                    value={10}
                        >
                                    10
                        </option>
                        <option
                                    value={25}
                        >
                                    25
                        </option>
                        <option
                                    value={50}
                        >
                                    50
                        </option>
            </select>
</div>,
          <div
            className="foodsListFilters__pages flex-right-end"
>
            <button
                        className="btn btn--page btn--disabled"
                        disabled={true}
                        onClick={[Function]}
            >
                        Prev
            </button>
            <button
                        className="btn btn--page btn--active"
                        disabled={true}
                        onClick={[Function]}
            >
                        1
            </button>
            <button
                        className="btn btn--page btn--disabled"
                        disabled={true}
                        onClick={[Function]}
            >
                        Next
            </button>
</div>,
        ],
        "className": "foodsListFilters",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="foodsListFilters__input"
              onChange={[Function]}
              placeholder="Search Foods"
              type="text"
              value=""
/>,
            "className": "foodsListFilters__search",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "foodsListFilters__input",
              "onChange": [Function],
              "placeholder": "Search Foods",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <select
                className="foodsListFilters__select"
                onChange={[Function]}
                value="usage"
>
                <option
                                value="usage"
                >
                                Usage
                </option>
                <option
                                value="date"
                >
                                Date
                </option>
</select>,
              <select
                className="foodsListFilters__select"
                onChange={[Function]}
                value={10}
>
                <option
                                value={1}
                >
                                1
                </option>
                <option
                                value={10}
                >
                                10
                </option>
                <option
                                value={25}
                >
                                25
                </option>
                <option
                                value={50}
                >
                                50
                </option>
</select>,
            ],
            "className": "foodsListFilters__options",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value="usage"
>
                    Usage
</option>,
                  <option
                    value="date"
>
                    Date
</option>,
                ],
                "className": "foodsListFilters__select",
                "onChange": [Function],
                "value": "usage",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Usage",
                    "value": "usage",
                  },
                  "ref": null,
                  "rendered": "Usage",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Date",
                    "value": "date",
                  },
                  "ref": null,
                  "rendered": "Date",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value={1}
>
                    1
</option>,
                  <option
                    value={10}
>
                    10
</option>,
                  <option
                    value={25}
>
                    25
</option>,
                  <option
                    value={50}
>
                    50
</option>,
                ],
                "className": "foodsListFilters__select",
                "onChange": [Function],
                "value": 10,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1",
                    "value": 1,
                  },
                  "ref": null,
                  "rendered": "1",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "10",
                    "value": 10,
                  },
                  "ref": null,
                  "rendered": "10",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "25",
                    "value": 25,
                  },
                  "ref": null,
                  "rendered": "25",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "50",
                    "value": 50,
                  },
                  "ref": null,
                  "rendered": "50",
                  "type": "option",
                },
              ],
              "type": "select",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="btn btn--page btn--disabled"
                disabled={true}
                onClick={[Function]}
>
                Prev
</button>,
              Array [
                <button
                  className="btn btn--page btn--active"
                  disabled={true}
                  onClick={[Function]}
>
                  1
</button>,
              ],
              <button
                className="btn btn--page btn--disabled"
                disabled={true}
                onClick={[Function]}
>
                Next
</button>,
            ],
            "className": "foodsListFilters__pages flex-right-end",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Prev",
                "className": "btn btn--page btn--disabled",
                "disabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Prev",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": 1,
                "className": "btn btn--page btn--active",
                "disabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": 1,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Next",
                "className": "btn btn--page btn--disabled",
                "disabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Next",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
